.dockerignore

```diff
@@ -0,0 +1,5 @@
+*
+!bin/run.sh
+!src/runner.nim
+!src/unittest_json.nim
+!tests
```

Comment:
Do we want to put the whole tests directory in the image? This won't have any benefit in production, and will just make the image bigger

Reply:
Indeed - let's not include it in the image. I'll work on it.

That's one of the things I was thinking when I wrote:
> Continue to work with the philosophy that the docker image should contain only the bare minimum.

Reply:
Possibilities:
1. Mount the `tests` directory
**Issue**: @ee7 had an issue with this because the container would modify file on the local file system when testing locally
**Solutions**: 
    1. Make the directory and files read-only 
          1. More steps to run locally
    2. Write a shell script that does it for us
          1. If it is run as the only CI step then less visibility as to where the job fails 
          2. If it is only for local testing then it adds an extra step for  maintenance of the CI script
     3. Write a yaml parser that runs the CI steps 
2. Copy them in as one of the steps in the CI

Reply:
> 2. Copy them in as one of the steps in the CI

I prefer this option and have implemented it for an upcoming PR. I think I could also mix it with

> ii. Write a shell script that does it for us

In a way that addresses 

> a. If it is run as the only CI step then less visibility as to where the job fails
> b. If it is only for local testing then it adds an extra step for maintenance of the CI script

But let's save it for another PR.

Reply:
Addressed in upcoming commit